cost/aws/elb/clb_unused/AWS_Delete_Unused_CLB.pt

```diff
@@ -0,0 +1,243 @@
+    var tags = clb['tags']
+    var isTagMatched=false
+    var tagKeyValue=""
+    for(var j=0; j < tags.length; j++){
+      tag = tags[j]
+      if(param_exclude_tags_lower.indexOf((tag['tagKey']+'='+tag['tagValue']).toLowerCase()) !== -1){
+        isTagMatched = true;
+      }
+      // Constructing tags with comma separated to display in detail_template
+      tagKeyValue = tagKeyValue + ', '+ tag['tagKey']+'='+tag['tagValue']
+    }
+
+    //If the CLB tag does not match with entered param_exclude_tags, then check if any instance exists in the CLB and check for instances health.
+    if(!(isTagMatched)){
+      clb_details={
+        loadBalancerName: clb['loadBalancerName'],
+        instanceId:(s_instanceIds.slice(2)),
+        availabilityZones:(s_availabilityZones.slice(2)),
+        tagKeyValue:(tagKeyValue.slice(2))
+      }
+      if(instancesList.length > 0){
+        var isInstanceNotHealthy=false
+        // Check if the CLB has no healthy instance with state and description matches for the values mentioned below
+		var state= "OutOfService"
+        var description= "Instance has failed at least the UnhealthyThreshold number of health checks consecutively."
+        for(var k=0; k < ds_aws_instances_health.length; k++){
+          instance= ds_aws_instances_health[k]
+          if(((instance['loadBalancerName']) == (clb['loadBalancerName'])) && (instance['state'] == state && instance['description'] == description)){
+            isInstanceNotHealthy=true
+          }
+        }
+        if(isInstanceNotHealthy){
+          content.push(clb_details)
+        }
+      }else{
+        content.push(clb_details)
+      }
+    }
+  }
+EOS
+end
+
+datasource "ds_unused_clb_map" do
+  run_script $js_aws_clb_filter_map, $ds_aws_clb_list_tags, $ds_aws_instances_health, $param_exclude_tags
+end
+
+policy "policy_delete_unused_CLB" do
+  validate $ds_unused_clb_map do
+  summary_template "List of Unused Elastic Load Balancers  ELB (CLB) in AWS"
+  detail_template <<-EOS
+## List of Unused Classic Load Balancers
+
+| Organization Name | LoadBalancer_Name | Instance_ID | Availability_Zones | TAG_Value |
+| ----------------- | ------------------ | ------------ | ----------------- | --------- |
+{{ range data -}}
+| {{rs_org_name}} | {{ .loadBalancerName }} | {{ .instanceId }} | {{ .availabilityZones }} | {{.tagKeyValue}} |
+{{ end -}}
+
+  ###### Policy Applied in Account: {{ rs_project_name }} (Account ID: {{ rs_project_id }}) within Org: {{ rs_org_name }} (Org ID: {{ rs_org_id }})
+  EOS
+  escalate $report_unused_CLB
+  escalate $approve_delete_CLB
+  check eq(size(data),0)
+  end
+end
+
+escalation "report_unused_CLB" do
+  email $param_email
+end
+
+escalation "approve_delete_CLB" do
+  request_approval  do
+    label "Approve Resource Deletion"
+    description "Approve escalation to run RightScale Cloud Workflow to delete unused CLB"
+    parameter "approval_reason" do
```

Comment:
Line indentation is not uniform in this escalation.

Reply:
Line indentation is taken care in this escalation.